src/ruby/lib/grpc/generic/rpc_desc.rb

```diff
@@ -72,18 +72,20 @@ def handle_bidi_streamer(active_call, mth)
       send_status(active_call, OK, 'OK', active_call.output_metadata)
     end
 
-    def run_server_method(active_call, mth)
-      # While a server method is running, it might be cancelled, its deadline
-      # might be reached, the handler could throw an unknown error, or a
-      # well-behaved handler could throw a StatusError.
-      if request_response?
-        handle_request_response(active_call, mth)
-      elsif client_streamer?
-        handle_client_streamer(active_call, mth)
-      elsif server_streamer?
-        handle_server_streamer(active_call, mth)
-      else  # is a bidi_stream
-        handle_bidi_streamer(active_call, mth)
+    def run_server_method(active_call, mth, interceptors = [])
+      intercept(interceptors, active_call, mth) do
```

Comment:
I believe this is what `go` does - each interceptor registers and has different methods that are called on the interceptor based on what type of request it is. We can go this route, if we'd prefer.